iceoryx_utils/test/moduletests/test_cxx_expected.cpp

```diff
@@ -50,247 +51,467 @@ class expected_test : public Test
             return m_a + m_b;
         }
 
+        bool operator==(const TestClass& rhs) const
+        {
+            return (m_a == rhs.m_a) && (m_b == rhs.m_b);
+        }
 
         int m_a;
         int m_b;
     };
 };
 
-TEST_F(expected_test, CreateWithValue)
+namespace iox
+{
+namespace cxx
+{
+template <>
+struct ErrorTypeAdapter<expected_test::TestClass>
+{
+    static expected_test::TestClass getInvalidState()
+    {
+        return expected_test::TestClass(-1, -1);
+    }
+};
+
+template <>
+struct ErrorTypeAdapter<std::string>
 {
-    auto sut = expected<int, float>::create_value(123);
+    static std::string getInvalidState()
+    {
+        return std::string("IAmInvalid");
+    }
+};
+} // namespace cxx
+} // namespace iox
+
+
+using TestClassAdapter = iox::cxx::ErrorTypeAdapter<expected_test::TestClass>;
+
+enum class TestError : uint8_t
+{
+    INVALID_STATE,
+    ERROR1,
+    ERROR2,
+    ERROR3
```

Comment:
Where is the creativity gone? I expected at least one hypnotoad and one Chuck Norris